src/modules/class-task-list.js

```diff
@@ -0,0 +1,76 @@
+import Task from './class-task.js';
+import isStorageAvailable from './local-storage-checker.js';
+
+const storageName = 'taskList';
+
+export default class TaskList {
+  constructor() {
+    this.data = [];
+    this.id = 1;
+    this.initUpdate();
+  }
+
+  initUpdate() {
+    if (isStorageAvailable('localStorage')) {
+      const data = JSON.parse(localStorage.getItem(storageName));
+
+      if (data && data.length !== 0) {
+        this.data = JSON.parse(localStorage.getItem(storageName));
+
+        const lastItem = this.data.reduce((prev, current) => {
+          const val = prev.id > current.id ? prev : current;
+          return val;
+        });
+        this.id = lastItem.id + 1;
+      }
+    }
+  }
+
+  addTask(task) {
+    if (isStorageAvailable) {
+      const taskObj = new Task(task, this.id);
+
+      this.data.push(taskObj);
+      localStorage.setItem(storageName, JSON.stringify(this.data));
+
+      this.id += 1;
+    }
+  }
+
+  removeTask(taskID) {
+    this.data = this.data.filter((i) => i.id !== taskID);
+    this.resetIds();
+    localStorage.setItem(storageName, JSON.stringify(this.data));
+  }
+
+  updateStatus(taskID, status) {
+    this.data = this.data.map((obj) => {
+      if (obj.id === taskID) {
+        return { ...obj, completed: status };
+      }
+
+      return obj;
+    });
+    localStorage.setItem(storageName, JSON.stringify(this.data));
+  }
+
+  renameTask(taskID, newTask) {
+    this.data = this.data.map((obj) => {
+      if (obj.id === taskID) {
+        return { ...obj, description: newTask };
+      }
+
+      return obj;
+    });
+    localStorage.setItem(storageName, JSON.stringify(this.data));
+  }
```

Comment:
- [ ] Kindly fix the editing functionality, I can edit an element in the browser, but that does not change the element in local storage, take a look at the image below to understand what I am saying.
- [ ] Let's fix the task completed feature as well, when a user clicks a check box, it should change a value of an element(task) from false to true, but currently, nothing happens as you can see on the image below.

![todolist](https://user-images.githubusercontent.com/53124548/167251397-5ac5cd6a-0060-4a00-91cd-824eb78cf14e.png)

Reply:
Hi @elisha2kyakpo1, I have to protest against your review and ask you to reconsider. it is saturday afternoon, there is still another project to do and you ask me to make changes that are not requirements to the task. I hope you undrestand the stress I am under right now.
kindly
Artur

Reply:
hi @elisha2kyakpo1, please ignore my last comment, I was very stressed. would you be kind enough as to giving me an idea on how to solve this problem?